### PR TITLE
Add option for custom generateSessionToken

### PR DESCRIPTION
**Pull request by halfmatthalfcat**
## ☕️ Reasoning

Instead of a random UUID, I want to leverage `cuid` as the session token. This PR allows the ability to define a custom `generateSessionToken` function and return the custom session token. Leverage original logic (random UUID) by default.

## 🧢 Checklist

- [x] Documentation
- [x] Ready to be merged

## 🎫 Affected issues

Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR.

Fixes: N/A

## 📌 Resources

- [Contributing guidelines](./CONTRIBUTING.md)
- [Code of conduct](./CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Comment by halfmatthalfcat** [comment]
@lluia @ThangHuuVu @ndom91 thoughts?

File: packages/next-auth/src/core/types.ts [hunk 1]
```diff
@@ -362,6 +362,7 @@ export interface CookiesOptions {
   pkceCodeVerifier: CookieOption
   state: CookieOption
   nonce: CookieOption
+  generateSessionToken: () => string
```
Comment (ThangHuuVu) [inline packages/next-auth/src/core/types.ts:365]: This `CookiesOptions` is more like for cookies-related configurations, as you can see they are all `CookieOption`
We should move this new config to somewhere else. I suggest it be inside of `session`.

**Review by ThangHuuVu — CHANGES_REQUESTED** [review]

File: packages/next-auth/src/core/types.ts (hunk 1 above)
Comment (halfmatthalfcat) [reply packages/next-auth/src/core/types.ts]: Ah good suggestion! Made the changes.

File: packages/next-auth/src/core/types.ts [hunk 2]
```diff
@@ -468,6 +468,13 @@ export interface SessionOptions {
    * @default 86400 // 1 day
    */
   updateAge: number
+  /**
+   * Generate a custom session token for database-based sessions.
+   * By default, a random UUID or string is generated depending on Node version
+   * however, you can specific your own custom string (such as cuids) to be used.
+   * @default randomUUID or randomBytes.toHex depending on Node version
```
Comment (balazsorban44) [inline packages/next-auth/src/core/types.ts:475]:
```suggestion
   * Generate a custom session token for database-based sessions.
   * By default, a random UUID or string is generated depending on the Node.js version.
   * However, you can specify your own custom string (such as CUID) to be used.
   * @default `randomUUID` or `randomBytes.toHex` depending on the Node.js version
```

File: packages/next-auth/src/core/init.ts [hunk 3]
```diff
@@ -86,6 +87,10 @@ export async function init({
       strategy: userOptions.adapter ? "database" : "jwt",
       maxAge,
       updateAge: 24 * 60 * 60,
+      generateSessionToken: () => {
+        // Use `randomUUID` if available. (Node 15.6++)
```
Comment (balazsorban44) [inline packages/next-auth/src/core/init.ts:91]:
```suggestion
        // Use `randomUUID` if available. (Node 15.6+)
```

**Pull request merged**